backends/tfhe-cuda-backend/cuda/src/integer/multiplication.cuh

```diff
@@ -88,7 +88,9 @@ void compress_device_array_with_map(cuda_stream_t *stream, Torus *src,
     auto cur_dst = &dst[total_copied * unit_size];
     auto cur_src = &src[s_index * unit_size];
     size_t copy_size = unit_size * number_of_unit * sizeof(Torus);
-    cuda_memcpy_async_gpu_to_gpu(cur_dst, cur_src, copy_size, stream);
+    if (copy_size > 0) {
+      cuda_memcpy_async_gpu_to_gpu(cur_dst, cur_src, copy_size, stream);
```

Comment:
What if we move this conditional into the `cuda_memcpy_*` methods? I think for all of them it would make sense to just return without doing anything if `copy_size == 0`. Wdyt?

Reply:
Yes, now you say it it seems obvious :sweat_smile: